Clang++11 emits lots of warnings about range analysis and misleading indentation

The following warnings are generated by Clang on master branch.

`/home/hbina/git/fatuous/thirdparty/assimp/code/Importer/IFC/IFCCurve.cpp:326:25: warning: loop variable 'sel' of type 'const Entry' (aka 'const shared_ptr<const STEP::EXPRESS::DataType>') creates a copy from type 'const Entry' [-Wrange-loop-analysis]
        for(const Entry sel :entity.Trim1) {
                        ^
/home/hbina/git/fatuous/thirdparty/assimp/code/Importer/IFC/IFCCurve.cpp:326:13: note: use reference type 'const Entry &' (aka 'const shared_ptr<const STEP::EXPRESS::DataType> &') to prevent copying
        for(const Entry sel :entity.Trim1) {
            ^~~~~~~~~~~~~~~~~
                        &
/home/hbina/git/fatuous/thirdparty/assimp/code/Importer/IFC/IFCCurve.cpp:343:25: warning: loop variable 'sel' of type 'const Entry' (aka 'const shared_ptr<const STEP::EXPRESS::DataType>') creates a copy from type 'const Entry' [-Wrange-loop-analysis]
        for(const Entry sel :entity.Trim2) {
                        ^
/home/hbina/git/fatuous/thirdparty/assimp/code/Importer/IFC/IFCCurve.cpp:343:13: note: use reference type 'const Entry &' (aka 'const shared_ptr<const STEP::EXPRESS::DataType> &') to prevent copying
        for(const Entry sel :entity.Trim2) {
            ^~~~~~~~~~~~~~~~~
                        &
/home/hbina/git/fatuous/thirdparty/assimp/code/XGL/XGLLoader.cpp:689:22: warning: loop variable 'p' has type 'const pairt &' (aka 'const pair<unsigned int, Assimp::XGLImporter::TempMaterialMesh> &') but is initialized with type 'std::pair<const unsigned int, Assimp::XGLImporter::TempMaterialMesh>' resulting in a copy [-Wrange-loop-analysis]
    for(const pairt& p : bymat) {
                     ^
/home/hbina/git/fatuous/thirdparty/assimp/code/XGL/XGLLoader.cpp:689:9: note: use non-reference type 'pairt' (aka 'pair<unsigned int, Assimp::XGLImporter::TempMaterialMesh>') to keep the copy or type 'const std::pair<const unsigned int, Assimp::XGLImporter::TempMaterialMesh> &' to prevent copying
    for(const pairt& p : bymat) {
`

I think I can fix this issue, and this is my first one in here :-3

Edit: err that error log didn't turn out the way its laid out but the problem is basically iterating over std::map, the pair types are <const Key, Value> not <Key, Value> so it will perform a temporary copy since its a reference.

Edit2: lol this copying was pretty bad 
```
struct TempMaterialMesh
    {
        TempMaterialMesh()
            : pflags()
            , matid()
        {}

        std::vector<aiVector3D> positions, normals;
        std::vector<aiVector2D> uvs;

        std::vector<unsigned int> vcounts;
        unsigned int pflags;
        unsigned int matid;
    };
```

--- a/code/COB/COBLoader.cpp
+++ b/code/COB/COBLoader.cpp
@@ -250,7 +250,7 @@ aiNode* COBImporter::BuildNodes(const Node& root,const Scene& scin,aiScene* fill
         const Mesh& ndmesh = (const Mesh&)(root);
         if (ndmesh.vertex_positions.size() && ndmesh.texture_coords.size()) {
 
-            typedef std::pair<unsigned int,Mesh::FaceRefList> Entry;
+            typedef std::pair<const unsigned int,Mesh::FaceRefList> Entry;
             for(const Entry& reflist : ndmesh.temp_map) {
                 {   // create mesh
                     size_t n = 0;

--- a/code/Importer/IFC/IFCCurve.cpp
+++ b/code/Importer/IFC/IFCCurve.cpp
@@ -323,7 +323,7 @@ public:
         // oh well.
         bool have_param = false, have_point = false;
         IfcVector3 point;
-        for(const Entry sel :entity.Trim1) {
+        for(const Entry& sel :entity.Trim1) {
             if (const ::Assimp::STEP::EXPRESS::REAL* const r = sel->ToPtr<::Assimp::STEP::EXPRESS::REAL>()) {
                 range.first = *r;
                 have_param = true;
@@ -340,7 +340,7 @@ public:
             }
         }
         have_param = false, have_point = false;
-        for(const Entry sel :entity.Trim2) {
+        for(const Entry& sel :entity.Trim2) {
             if (const ::Assimp::STEP::EXPRESS::REAL* const r = sel->ToPtr<::Assimp::STEP::EXPRESS::REAL>()) {
                 range.second = *r;
                 have_param = true;

--- a/code/MDL/MDLLoader.cpp
+++ b/code/MDL/MDLLoader.cpp
@@ -1421,7 +1421,7 @@ void MDLImporter::InternReadFile_3DGS_MDL7( )
         avOutList[i].reserve(3);
 
     // buffer to held the names of all groups in the file
-	const size_t buffersize( AI_MDL7_MAX_GROUPNAMESIZE*pcHeader->groups_num );
+    const size_t buffersize(AI_MDL7_MAX_GROUPNAMESIZE*pcHeader->groups_num);
 	char* aszGroupNameBuffer = new char[ buffersize ];
 
     // read all groups

--- a/code/MDL/MDLLoader.cpp
+++ b/code/MDL/MDLLoader.cpp
@@ -1422,10 +1422,10 @@ void MDLImporter::InternReadFile_3DGS_MDL7( )
 
     // buffer to held the names of all groups in the file
     const size_t buffersize(AI_MDL7_MAX_GROUPNAMESIZE*pcHeader->groups_num);
-	char* aszGroupNameBuffer = new char[ buffersize ];
+    char* aszGroupNameBuffer = new char[ buffersize ];
 
     // read all groups
-    for (unsigned int iGroup = 0; iGroup < (unsigned int)pcHeader->groups_num;++iGroup) {
+    for (unsigned int iGroup = 0; iGroup < (unsigned int)pcHeader->groups_num; ++iGroup) {
         MDL::IntGroupInfo_MDL7 groupInfo((BE_NCONST MDL::Group_MDL7*)szCurrent,iGroup);
         szCurrent = (const unsigned char*)(groupInfo.pcGroup+1);
 

--- a/code/XGL/XGLLoader.cpp
+++ b/code/XGL/XGLLoader.cpp
@@ -685,7 +685,7 @@ bool XGLImporter::ReadMesh(TempScope& scope)
     }
 
     // finally extract output meshes and add them to the scope
-    typedef std::pair<unsigned int, TempMaterialMesh> pairt;
+    typedef std::pair<const unsigned int, TempMaterialMesh> pairt;
     for(const pairt& p : bymat) {
         aiMesh* const m  = ToOutputMesh(p.second);
         scope.meshes_linear.push_back(m);

